### DISCUSSION TITLE
Ability to change the sweep time.

### DISCUSSION THREAD

**Discussion by Rappoman** [ESPresense/ESPresense#102] [Support]
First off a really good job on the implementation, you have made it easy to set-up and use.  

Would it be possible to add an option to set (reduce) the time between sweeps.  This would allow for the base stations to identify room changes quicker.  This in turn would help with automations in HA.

Thanks in advance.

**Comment by DTTerastar** [comment]
I'm not sure which value you mean.  We use a continuous scan.

**Comment by Rappoman** [comment]
Thanks for the quick reply.  I am not even sure if there is a value/setting that can be altered.  But I notice when moving through the house and going from one room to another it can take several (10+) seconds for Home Assistant to update the room a device is in.

**Comment by DTTerastar** [comment]
Some devices don't broadcast very often.  Tiles are like every 4 seconds, so they're slow.  Apple Watches are 10x a second so they move quick.  What kind of device are you tracking?

**Comment by Rappoman** [comment]
I am using several at the moment, Samsung S10 (HA Android app), Fitbit Ionic, Fitbit Charge 3 and another cheap sports watch.

**Comment by DTTerastar** [comment]
All of those take 10 seconds to be found in the new room?

**Comment by Rappoman** [comment]
Slight change of subject, I have been doing some testing and it would seem that the quality of the usb cable supplying power to the ESP32 makes quite a bit of difference to the distance reading and the accuracy of the base station correctly identifying the room a ble device is in.  

I can also see what you are saying about the the frequency of broadcast between the different devices I have.  The Fitbit Ionic and S10 appear to be the ones that broadcast less frequently at about 10 - 12 seconds.  The other 2 devices about 6 - 8 seconds.

Thanks for your assistance.

**Comment by DTTerastar** [comment]
What hardware are you using?  I wouldn't expect usb cabling to make a difference.  The positioning should tho.  If you install MQTT Explorer you can graph the "raw" and "distance" values.  Raw is the unfiltered distance, and distance has gone through 2 filters.  Do you see raw responding any faster?

**Reply by Rappoman** [reply]
I am using ESP32 development boards.  I installed MQTT  Explorer.  I would not say that either updates faster than the other.  However, the raw does appear to show as being closer in general.

**Comment by DTTerastar** [comment]
Raw should be less "noisy".  It should lag by a little bit since it's trading a little lag for stability.
